src/app/delegation_backend/src/itn_uptime_analyzer/uptime.go

```diff
@@ -0,0 +1,346 @@
+					}
+
+					defer objHandle.Body.Close()
+
+					objContents, err := io.ReadAll(objHandle.Body)
+					if err != nil {
+						log.Fatalf("Error getting creating reader for json: %v\n", err)
+					}
+
+					err = json.Unmarshal(objContents, &submissionDataToday)
+					if err != nil {
+						log.Fatalf("Error unmarshaling bucket content: %v\n", err)
+					}
+
+					if submissionDataToday.GraphqlControlPort != 0 {
+						if (identity["public-key"] == submissionDataToday.Submitter.String()) && (identity["public-ip"] == submissionDataToday.RemoteAddr) && (identity["graphql-port"] == strconv.Itoa(submissionDataToday.GraphqlControlPort)) {
+							if lastSubmissionTimeString != "" {
+								lastSubmissionTime, err = time.Parse(time.RFC3339, lastSubmissionTimeString)
+								if err != nil {
+									log.Fatalf("Error parsing time: %v\n", err)
+								}
+							}
+
+							currentSubmissionTime, err := time.Parse(time.RFC3339, submissionDataToday.CreatedAt)
+							if err != nil {
+								log.Fatalf("Error parsing time: %v\n", err)
+							}
+
+							if (lastSubmissionTimeString != "") && (currentSubmissionTime.After(lastSubmissionTime.Add(time.Duration(syncPeriod-5) * time.Minute))) && (currentSubmissionTime.Before(lastSubmissionTime.Add(time.Duration(syncPeriod+5) * time.Minute))) {
+								uptimeToday = append(uptimeToday, true)
+								lastSubmissionTimeString = submissionDataToday.CreatedAt
+							} else if lastSubmissionTimeString == "" {
+								uptimeToday = append(uptimeToday, true)
+								lastSubmissionTimeString = submissionDataToday.CreatedAt
+							}
+
+						}
+					} else {
+						if (identity["public-key"] == submissionDataToday.Submitter.String()) && (identity["public-ip"] == submissionDataToday.RemoteAddr) {
+							if lastSubmissionTimeString != "" {
+								lastSubmissionTime, err = time.Parse(time.RFC3339, lastSubmissionTimeString)
+								if err != nil {
+									log.Fatalf("Error parsing time: %v\n", err)
+								}
+							}
+
+							currentSubmissionTime, err := time.Parse(time.RFC3339, submissionDataToday.CreatedAt)
+							if err != nil {
+								log.Fatalf("Error parsing time: %v\n", err)
+							}
+
+							if (lastSubmissionTimeString != "") && (currentSubmissionTime.After(lastSubmissionTime.Add(time.Duration(syncPeriod-5) * time.Minute))) && (currentSubmissionTime.Before(lastSubmissionTime.Add(time.Duration(syncPeriod+5) * time.Minute))) {
+								uptimeToday = append(uptimeToday, true)
+								lastSubmissionTimeString = submissionDataToday.CreatedAt
+							} else if lastSubmissionTimeString == "" {
+								uptimeToday = append(uptimeToday, true)
+								lastSubmissionTimeString = submissionDataToday.CreatedAt
+							}
+
+						}
+					}
+				}
+			}
+
+			uptimePercent := (float64(len(uptimeToday)+len(uptimeYesterday)) / float64(numberOfSubmissionsNeeded)) * 100
+			if uptimePercent > 100.00 {
+				uptimePercent = 100.00
+			}
+			identity["uptime"] = fmt.Sprintf("%.2f%%", uptimePercent)
```

Comment:
Certainly a bug here, it will be called after reading the first page of yesterday and all pages of today

I'm sure there is some issue with `}` braces order